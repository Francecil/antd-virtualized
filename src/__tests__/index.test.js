const OLD_NODE_ENV = process.env.NODE_ENV;
process.env.NODE_ENV = 'development';
const warnSpy = jest.spyOn(console, 'warn').mockImplementation(() => { });
const antdv = require('..');

describe('antdv', () => {
  afterAll(() => {
    process.env.NODE_ENV = OLD_NODE_ENV;
  });

  it('exports modules correctly', () => {
    console.log(antdv)
    expect(Object.keys(antdv)).toMatchSnapshot();
  });

  it('should hint when import all components in dev mode', () => {
    // 通过是否调用 console.warn(xxxx) 进行判断
    expect(warnSpy).toHaveBeenCalledWith(
      'You are using a whole package of antd virtualized, please use https://www.npmjs.com/package/babel-plugin-import to reduce app bundle size.',
    );
    warnSpy.mockRestore();
  });
});