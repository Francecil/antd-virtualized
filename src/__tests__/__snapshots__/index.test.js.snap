// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`antdv exports modules correctly 1`] = `
Array [
  "Select",
  "TreeSelect",
  "Tree",
  "Version",
]
`;
